central/reports/service/v2/service_impl.go

```diff
@@ -0,0 +1,54 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/stackrox/rox/central/reportconfigurations/service/common"
+	metadataDS "github.com/stackrox/rox/central/reports/metadata/datastore"
+	apiV2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/logging"
+	"google.golang.org/grpc"
+)
+
+const maxPaginationLimit = 1000
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type serviceImpl struct {
+	metadataDatastore metadataDS.DataStore
+}
+
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
+		apiV2.RegisterReportServiceServer(grpcServer, s)
+	}
+}
+
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
+		return apiV2.RegisterReportConfigurationServiceHandler(ctx, mux, conn)
+	}
+	return nil
+}
+
+func (*serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, common.Authorizer.Authorized(ctx, fullMethodName)
```

Comment:
This is using the ReportConfigurationService's authorizer, but that will not be able to authorize the requests for `ReportService` . So you would need to create a similar authorizer for ReportService.